Cubical/Experiments/truncations.agda

```diff
@@ -0,0 +1,165 @@
+{-# OPTIONS --cubical --safe #-}
+module Cubical.Experiments.truncations where
+
+open import Cubical.Data.Nat.Base
+open import Cubical.Core.Prelude
+
+isOfHLevel : ∀ {ℓ} → ℕ → Set ℓ → Set ℓ
+isOfHLevel 0 A = isContr A
+isOfHLevel 1 A = isProp A
+isOfHLevel (suc (suc n)) A = (x y : A) → isOfHLevel (suc n) (x ≡ y)
+
+isGroupoid : Set → Set
+isGroupoid = isOfHLevel 3
+
+isTwoGroupoid : Set → Set
+isTwoGroupoid = isOfHLevel 4
+
+transport' : (p : I → Set) → (p i0) → (p i1)
+transport' p a = transp (λ i → p i) i0 a
+
+data propTrunc (A : Set) : Set where
+  pinc : A → propTrunc A
+  pIsProp : isProp (propTrunc A)
+
+propTruncElim : (A : Set) (B : (propTrunc A) → Set)
+                (bP : (x : propTrunc A) → isProp (B x))
+                (f : (x : A) → B (pinc x))
+                (x : propTrunc A) → B x
+propTruncElim A B bP f = proof
+  where
+  proof : (x : propTrunc A) → B x
+  proof (pinc x) = f x
+  proof (pIsProp x y i) = L i
+    where
+    pA = propTrunc A
+    L0 = bP x
+    C  = pIsProp x y
+    B' : {x y : pA} → (p : x ≡ y) → Set
+    B' {x} {y} p = (a : B x) → (b : B y) → PathP (λ i → B (p i)) a b 
+    B0 : B' (λ j → C i0) ≡ B' (λ j → C j)
+    B0 i = B' (λ j → C (i ∧ j))
+    L1 : B' C
+    L1 = transport B0 L0
+    L : PathP (λ i → B (pIsProp x y i)) (proof x) (proof y)
+    L = L1 (proof x) (proof y)
+
+data setTrunc (A : Set) : Set where
+  tinc : A → setTrunc A
+  tIsSet : isSet (setTrunc A)
+
+setTruncElim : (A : Set) (B : (setTrunc A) → Set)
```

Comment:
Could you compress this proof a bit? It is quite hard to read because of all the local definitions with very short names.

I would like to understand how it compares to https://github.com/agda/cubical/blob/master/Cubical/HITs/SetTruncation/Properties.agda#L37 . Are the proofs the same or not? Note that J is just a transport with a `/\` connection https://github.com/agda/cubical/blob/master/Cubical/Core/Prelude.agda#L88